4-WebApp-your-API/4-1-MyOrg/AppCreationScripts-withCert/Configure.ps1

```diff
@@ -148,228 +144,208 @@ Function CreateAppRole([string] $types, [string] $name, [string] $description)
    Write-Host "Creating the AAD application (TodoListClient-aspnetcore-webapi)"
+   
    # create the application 
-   $clientAadApplication = New-AzureADApplication -DisplayName "TodoListClient-aspnetcore-webapi" `
-                                                  -HomePage "https://localhost:44321/" `
-                                                  -LogoutUrl "https://localhost:44321/signout-oidc" `
-                                                  -ReplyUrls "https://localhost:44321/", "https://localhost:44321/signin-oidc" `
-                                                  -IdentifierUris "https://$tenantName/TodoListClient-aspnetcore-webapi" `
-                                                  -PublicClient $False
-
-   # Generate a certificate
-   Write-Host "Creating the client application (TodoListClient-aspnetcore-webapi)"
-
-   $isOpenSSL = Read-Host ' By default certificate is generated using New-SelfSignedCertificate. Do you want to generate cert using OpenSSL(Y/N)?'
-
-   if($isOpenSSL -eq 'Y')
-   {
-        $certificate=openssl req -x509 -newkey rsa:4096 -sha256 -days 365 -keyout webapp.key -out webapp.cer -nodes -batch
-        openssl pkcs12 -export -out webapp.pfx -inkey webapp.key -in webapp.cer
-   }
-   else
-   {
-        $certificate=New-SelfSignedCertificate -Subject webapp `
+   $clientAadApplication = New-MgApplication -DisplayName "TodoListClient-aspnetcore-webapi" `
+                                                      -Web `
+                                                      @{ `
+                                                          RedirectUris = "https://localhost:44321/", "https://localhost:44321/signin-oidc"; `
+                                                          HomePageUrl = "https://localhost:44321/"; `
+                                                          LogoutUrl = "https://localhost:44321/signout-oidc"; `
+
+                                                          ImplicitGrantSettings = @{EnableIdTokenIssuance=$true;EnableAccessTokenIssuance=$false} ` 
+ 
+                                                        } `
+                                                      -SignInAudience AzureADMyOrg `
+                                                      #-PublicClient $False
+    $tenantName = (Get-MgApplication -ApplicationId $clientAadApplication.Id).PublisherDomain
+    Update-MgApplication -ApplicationId $clientAadApplication.Id -IdentifierUris @("https://$tenantName/TodoListClient-aspnetcore-webapi")
+    
+    # Generate a certificate
+    Write-Host "Creating the client application (TodoListClient-aspnetcore-webapi)"
+
+    $certificateName = 'TodoListClient-aspnetcore-webapi'
+
+    # temporarily disable the option and procees to certificate creation
+    #$isOpenSSL = Read-Host ' By default certificate is generated using New-SelfSignedCertificate. Do you want to generate cert using OpenSSL(Y/N)?'
+    $isOpenSSl = 'N'
+    if($isOpenSSL -eq 'Y')
```

Comment:
Wouldn't this never run?

Reply:
it was disabled temporary, I will add it in later versions